### PR TITLE
infra: Dockefile for custom groovy image

### PR DESCRIPTION
**Pull request by AkMo3**

### CODE REVIEW

File: devops/docker/Groovy/Dockerfile
Comment (strkkk) [inline devops/docker/Groovy/Dockerfile:67]: this is a copy-paste of official groovy image dockerfile.

It is way easier to make it like
```
FROM <groovy image name>
RUN <installation of git and whatever we need>
```
it should work as well

**Review by strkkk — CHANGES_REQUESTED** [review]

File: devops/docker/Groovy/Dockerfile
Comment (AkMo3) [reply devops/docker/Groovy/Dockerfile:23]: Done

**Comment by romani** [comment]
@strkkk , please finish review

File: devops/docker/Groovy/Dockerfile
Comment (strkkk) [inline devops/docker/Groovy/Dockerfile:25]: I believe this (RUN) and previous (USER) steps are not needed, please remove them

**Review by strkkk — CHANGES_REQUESTED** [review]

File: devops/docker/Groovy/Dockerfile
Comment (AkMo3) [reply devops/docker/Groovy/Dockerfile:25]: @strkkk Both these `USER` changes are essential. Without a change to the root user, one cannot get access to the protected files of the container. You may confirm through the below logs. The second one is important since after installing new packages, we need to sure that the entire container runs as expected in non-root environments after all the changes. 

```
$ cat Dockerfile 
#################### Using official groovy image ####################
FROM groovy:jdk11
#################### End of images used ############################# 

##################### Installing Maven and git #######################
RUN apt-get update \
    && echo "Installing maven and git dependencies" \
    && apt-get install --yes --no-install-recommends \
        git \
        maven
##################### End of install #################################

##################### Check install of groovy ########################
RUN set -o errexit -o nounset \
    && echo "Testing Groovy installation" \
    && groovy --version
##################### End of Dockerfile ##############################

$ docker build . -t akmo03/groovy-jdk11:latest
Sending build context to Docker daemon   2.56kB
Step 1/3 : FROM groovy:jdk11
 ---> b55bd63dcb2d
Step 2/3 : RUN apt-get update     && echo "Installing maven and git dependencies"     && apt-get install --yes --no-install-recommends         git         maven
 ---> Running in 1b3c4c387941
Reading package lists...
E: List directory /var/lib/apt/lists/partial is missing. - Acquire (13: Permission denied)
The command '/bin/sh -c apt-get update     && echo "Installing maven and git dependencies"     && apt-get install --yes --no-install-recommends         git         maven' returned a non-zero code: 100
```

Comment (strkkk) [reply devops/docker/Groovy/Dockerfile:25]: yes, I understand that. I mean that only last user command (switch to groovy) is not really needed, it is only to run test installation command.

I mean to delete run command and user switch which is just above it

Comment (AkMo3) [reply devops/docker/Groovy/Dockerfile]: Done. I think it is usually a standard to test new images, but we skip through that since we using the official image :)

Comment (strkkk) [reply devops/docker/Groovy/Dockerfile]: you always can make a local test. Build a new image and just run some command

**Review by strkkk — APPROVED** [review]

**Comment by strkkk** [comment]
@AkMo3 @romani I am ok with this docker file. CI failure is not related to changes.
We can push this image to checkstyle repo and try to use it in pitest stuff.

File: devops/docker/Groovy/Dockerfile
Comment (romani) [inline devops/docker/Groovy/Dockerfile:2]: we are on jdk8 for now

**Review by romani — CHANGES_REQUESTED** [review]
item:

**Comment by romani** [comment]
@AkMo3 , please investigate apapveyor failure, it is caused by update https://github.com/checkstyle/checkstyle/commit/e4478c1adbc89fec05e54f5652957f4ead497978

I suspecting that appveyor is chekouting code and update line ending to "\r\n", but in xpath we check only for "//\n".
It is strange that this was not detected in main repo appveyor build. Please compare their code checkout process.
Chekcstye config in main repo also need to be updated allow "\r" as some user might have problems on Windows also.

**Comment by AkMo3** [comment]
> It is strange that this was not detected in main repo appveyor build.

In the main repo, we have fixed line ending to be `lf` only,

https://github.com/checkstyle/checkstyle/blob/e59f2d2386e6d456bc8f58e2054fd2712a023327/.gitattributes#L2 

But, in this repo, we do not have such configuration, so after checkout, as appveyor runs in windows, git doesn't changes the line ending and keeps it `clrf` only.

File: devops/docker/Groovy/Dockerfile
Comment (AkMo3) [reply devops/docker/Groovy/Dockerfile]: Done

**Comment by romani** [comment]
we already have image https://github.com/checkstyle/contribution/blob/e312aac6a52d916b55b257c3d30832a673558740/devops/docker/maven-build-image/jdk-8u162b12/Dockerfile
for our builds

why it is not enough ? or ?

File: devops/docker/Groovy/Dockerfile
Comment (romani) [inline devops/docker/Groovy/Dockerfile:1]: make path:

no-exception-bulid-image/jdk-8-groovy-git-mvn/Dockerfile

**Review by romani — CHANGES_REQUESTED** [review]
Please in separate PR suppress violation from problematic Check - to let CI pass.
after merge and send one more PR to remove this suppression - I will merge it evenually  after deploy.

item:

File: devops/docker/Groovy/Dockerfile
Comment (AkMo3) [reply devops/docker/Groovy/Dockerfile]: Please see #563.

**Review by romani — APPROVED** [review]

**Comment by romani** [comment]
I need to build docker image and push to dockerhub .... To let us reuse official image

**Pull request merged**
